apps/condo/domains/notification/tasks.js

```diff
@@ -100,6 +100,27 @@ async function deliverMessage (messageId) {
         sender: message.sender,
     }
 
+    const blackListWhere = {
+        OR: [
+            message.organization && { organization: { id: message.organization.id } },
+            message.user && { user: { id: message.user.id } },
+            message.phone && { phone: message.phone },
+            message.email && { email: message.email },
+        ].filter(Boolean),
+    }
+
+    const blackListRules = await MessageBlackList.getAll(keystone, blackListWhere)
+    if (!isEmpty(blackListRules)) {
+        const isMessageTypeInBlackList = blackListRules.find(rule => rule.type === message.type || isEmpty(rule.type))
+
+        if (isMessageTypeInBlackList) {
+            return await Message.update(keystone, message.id, {
+                ...baseAttrs,
+                status: MESSAGE_ERROR_STATUS,
```

Comment:
Is there a way to tell the reason of this error as presence in a black list?

Reply:
added error message in processingMeta of Message